### PR TITLE
[DR] Advance token in case of missing keys

### PR DESCRIPTION
**Pull request by snalli**
The replication loop waits for missing keys to be sent by remote replica so that it can apply them locally. The case for BackupChecker is different. It needs to report those missing keys and just move on.

### CODE REVIEW

**Review by JingQianCloud — COMMENTED** [review]
Overall LGTM.
There is one binary file with weird name in the PR. I assume it was by accident.

**Comment by snalli** [comment]
> Overall LGTM. There is one binary file with weird name in the PR. I assume it was by accident.

yeah just an error

File: ambry-replication/src/main/java/com/github/ambry/replication/BackupCheckerThread.java [hunk 1]
```diff
@@ -168,23 +185,23 @@ protected void fixMissingStoreKeys(
    * @param acceptableStoreErrorCodes Acceptable error codes when retrieving the blob from local-store
```
Comment (ankagrawal) [inline ambry-replication/src/main/java/com/github/ambry/replication/BackupCheckerThread.java:185]: nit: you need to edit javadocs.

Comment (ankagrawal) [reply ambry-replication/src/main/java/com/github/ambry/replication/BackupCheckerThread.java:185]: Why do you need `MessageInfoType` when `MessageInfo` is already passed as argument.

File: ambry-replication/src/main/java/com/github/ambry/replication/BackupCheckerThread.java [hunk 2]
```diff
@@ -150,16 +150,33 @@ protected void fixMissingStoreKeys(
...
+    for (int i = 0; i < exchangeMetadataResponseList.size(); i++) {
+      ExchangeMetadataResponse exchangeMetadataResponse = exchangeMetadataResponseList.get(i);
+      RemoteReplicaInfo remoteReplicaInfo = replicasToReplicatePerNode.get(i);
       if (exchangeMetadataResponse.serverErrorCode == ServerErrorCode.No_Error) {
         for (MessageInfo messageInfo: exchangeMetadataResponse.getMissingStoreMessages()) {
           // Check local store once before logging an error
-          checkLocalStore(messageInfo, replicasToReplicatePerNode.get(0), acceptableLocalBlobStates, acceptableStoreErrorCodes);
+          checkLocalStore(messageInfo, replicasToReplicatePerNode.get(0), acceptableLocalBlobStates, acceptableStoreErrorCodes, MessageInfoType.PUT);
         }
+        // Advance token so that we make progress in spite of missing keys,
```
Comment (ankagrawal) [inline ambry-replication/src/main/java/com/github/ambry/replication/BackupCheckerThread.java:161]: Thanks for adding the comment here. This is very helpful

**Review by ankagrawal — COMMENTED** [review]
looks good overall. A couple of nitpicks.

File: ambry-replication/src/main/java/com/github/ambry/replication/BackupCheckerThread.java (hunk 1 above)
Comment (snalli) [reply ambry-replication/src/main/java/com/github/ambry/replication/BackupCheckerThread.java]: 1/ what shd i add to javadocs ?
2/ yeah, removed.

**Review by ankagrawal — APPROVED** [review]

**Pull request merged**